libp2p/network/url_conversion_test.go

```diff
@@ -63,40 +65,37 @@ func TestPeerMultiAddrFromServiceURL_Error(t *testing.T) {
 	hostnames := map[string]string{
 		fqdn: "www.google.com",
 		ip4:  "142.250.181.196",
-		ip6:  "2a00:1450:4005:802::2004",
+		ip6:  "[2a00:1450:4005:802::2004]",
 	}
 
 	testCases := []struct {
 		name             string
 		serviceURLFormat string
-		// TECHDEBT: assert specific errors.
-		expectedErrContains string
+		expectedErr      string
```

Comment:
Thanks for tackling this! :raised_hands: 

Once we have error message vars for each error (above), we can de-duplicate the string literals here.